plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/ContributingModule.java

```diff
@@ -187,5 +195,7 @@ public IWorkspaceRoot get() {
 		binder.bind(NpmBinary.class).in(SINGLETON);
 		binder.bind(NpmrcBinary.class).in(SINGLETON);
 
+		binder.bind(TypesKeywordProvider.class).in(SINGLETON);
```

Comment:
We seem to be mixing ways to declare singleton. In some cases we use `@Singleton` and sometimes `in(Scopes.SINGLETON)`. Personally I prefer annotation, as it is on declaration site, not in some config you hardly ever see.
Could we just use annotation in all places?

for reference https://github.com/google/guice/wiki/Scopes#eager-singletons

Reply:
Good point. I change all occurrences to use the annotation and also add a comment that states that only annotations are validated.